crates_io_worker/src/background_job.rs

```diff
@@ -27,8 +26,8 @@ pub trait BackgroundJob: Serialize + DeserializeOwned + Send + Sync + 'static {
     /// The application data provided to this job at runtime.
     type Context: Clone + Send + 'static;
 
-    /// Execute the task. This method should define its logic
-    async fn run(&self, ctx: Self::Context) -> anyhow::Result<()>;
+    /// Execute the task. This method should define its logic.
+    fn run(&self, ctx: Self::Context) -> impl Future<Output = anyhow::Result<()>> + Send;
```

Comment:
`async fn` is apparently not possible here yet due to the necessary `Send` bound